### PR TITLE
[ci-visibility] Fix issues with early flake detection in jest

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Remove the `.trim()` in [this line](https://github.com/DataDog/dd-trace-js/pull/4183/files#diff-74a96f6d5ac2137dcdae2e86565193c9f9a46b10276918a309dd19f974da1353L159), since it was buggy. It was there to remove leading empty spaces in the case where the test case had no `describe` clause:
```javascript
`${describeSuffix} ${event.testName}`
```

But this also modifies the test name, in case it includes a trailing empty space: 

```javascript
describe('context', () => {
  test('trailing space ', () => {...})
})
```

Which causes the early flake detection algorithm to fail. 

### Motivation
Fix early flake detection. 

### Plugin Checklist

- [x] Unit tests.

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: packages/dd-trace/src/plugins/ci_plugin.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ module.exports = class CiPlugin extends Plugin {
       }
       this.tracer._exporter.getLibraryConfiguration(this.testConfiguration, (err, libraryConfig) => {
         if (err) {
-          log.error(`Intelligent Test Runner configuration could not be fetched. ${err.message}`)
+          log.error(`Library configuration could not be fetched. ${err.message}`)
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/ci_plugin.js:45]: unrelated fix

**Review by rochdev — APPROVED** [review]

**Review by romainkomorndatadog — APPROVED** [review]
